src/include/gurt/atomic.h

```diff
@@ -43,7 +43,18 @@
 #if HAVE_STDATOMIC
 
 #include <stdatomic.h>
+
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#ifdef __INTEL_COMPILER
+#define ATOMIC volatile
```

Comment:
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt